src/main/java/duke/UserInterface.java

```diff
@@ -0,0 +1,533 @@
+     *                   Task that must be added to the list of Tasks. Supported tasks: todo, deadline, event
+     */
+    private void addTask(ArrayList<String> inputTasks) {
+        try {
+            printDivider();
+            this.tasks.add(buildTask(inputTasks));
+            System.out.println("Got it. I've added this task:\n  " + this.tasks.get(this.tasks.size() - 1));
+            System.out.println("Now you have " + tasks.size()
+                    + (tasks.size() == 1 ? " task" : " tasks") + " in the list.");
+            printDivider();
+        } catch (Exception e) {
+            System.out.println("Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Constructs a subclass of Task from an ArrayList<String>.
+     *
+     * @param taskStringArray the ArrayList<String> containing the user's input, split up by whitespace.
+     *                        The first element of this list should contain the subclass of task to be constructed.
+     * @return a subclass of Task determined by the first element of taskStringArray.
+     */
+    private Task buildTask(ArrayList<String> taskStringArray) throws DukeException {
+        String taskType = taskStringArray.get(0);
+        ArrayList<StringBuilder> taskParts = splitTask(taskStringArray);
+        validateTask(taskType, taskParts);
+        String prepositions = String.valueOf(taskParts.get(taskParts.size() - 1));
+        switch (taskType) {
+        case COMMAND_TODO:
+            return new ToDo(String.valueOf(taskParts.get(0)));
+        case COMMAND_DEADLINE:
+            return new Deadline(String.valueOf(taskParts.get(0)),
+                    String.valueOf(taskParts.get(1)),
+                    prepositions);
+        case COMMAND_EVENT:
+            return new Event(String.valueOf(taskParts.get(0)),
+                    String.valueOf(taskParts.get(1)),
+                    prepositions);
+        }
+        return null;
+    }
+
+    /**
+     * Validates the input of the task to check for user error.
+     *
+     * @param taskType  the type of task (todo, deadline etc).
+     * @param taskParts the list of components that form the task.
+     * @throws DukeException error message if task is invalid
+     */
+    private void validateTask(String taskType, ArrayList<StringBuilder> taskParts) throws DukeException {
+        if (taskParts.isEmpty()) {
+            throw new DukeException("The description of a " + taskType + " cannot be empty.");
+        }
+        String prepositions = String.valueOf(taskParts.get(taskParts.size() - 1));
+        switch (taskType) {
+        case COMMAND_TODO:
+            if (!prepositions.isEmpty()) {
+                throw new DukeException("Todo tasks cannot include the preposition " + prepositions +" in the description.");
+            }
+            break;
+        case COMMAND_DEADLINE:
+            if (taskParts.get(0).toString().equals("")
+                    || taskParts.get(1).toString().equals("")) {
```

Comment:
This is a [compound boolean expression]([url](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions)) which could be refactored in intermediate steps with boolean names.